modules/nextflow/src/main/groovy/nextflow/cli/CleanImpl.groovy

```diff
@@ -41,58 +40,39 @@ import nextflow.trace.TraceRecord
 import nextflow.util.HistoryFile.Record
 
 /**
- * Implements cache clean up command
+ * CLI `clean` sub-command
  *
  * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
  * @author Lorenz Gerber <lorenzottogerber@gmail.com>
  */
 @Slf4j
 @CompileStatic
-@Parameters(commandDescription = "Clean up project cache and work directories")
-class CmdClean extends CmdBase implements CacheBase {
-
-    static final public NAME = 'clean'
-
-    @Parameter(names=['-q', '-quiet'], description = 'Do not print names of files removed', arity = 0)
-    boolean quiet
-
-    @Parameter(names=['-f', '-force'], description = 'Force clean command', arity = 0)
-    boolean force
-
-    @Parameter(names=['-n', '-dry-run'], description = 'Print names of file to be removed without deleting them' , arity = 0)
-    boolean dryRun
-
-    @Parameter(names='-after', description = 'Clean up runs executed after the specified one')
-    String after
-
-    @Parameter(names='-before', description = 'Clean up runs executed before the specified one')
-    String before
-
-    @Parameter(names='-but', description = 'Clean up all runs except the specified one')
-    String but
-
-    @Parameter(names=['-k', '-keep-logs'], description = 'Removes only temporary files but retains execution log entries and metadata')
```

Comment:
Looks smart, but not getting where these are implemented

Reply:
They are implemented in `cli.v1.CleanCmd` and `cli.v2.CleanCmd` respectively.